lib/Doctrine/Common/Lexer/Token.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Common\Lexer;
+
+use ArrayAccess;
+use Countable;
+use Doctrine\Deprecations\Deprecation;
+use ReturnTypeWillChange;
+
+use function in_array;
+
+/**
+ * @template T of string|int
+ * @implements ArrayAccess<string,mixed>
+ */
+final class Token implements ArrayAccess, Countable
+{
+    /**
+     * The string value of the token in the input string
+     *
+     * @readonly
+     * @var string|int
+     */
+    public $value;
+
+    /**
+     * The type of the token (identifier, numeric, string, input parameter, none)
+     *
+     * @readonly
+     * @var T|null
+     */
+    public $type;
+
+    /**
+     * The position of the token in the input string
+     *
+     * @readonly
+     * @var int
+     */
+    public $position;
+
+    /**
+     * @param string|int $value
+     * @param T|null     $type
+     */
+    public function __construct($value, $type, int $position)
+    {
+        $this->value    = $value;
+        $this->type     = $type;
+        $this->position = $position;
+    }
+
+    /** @param T $types */
```

Comment:
```suggestion
    /** @param T ...$types */
```

see https://phpstan.org/writing-php-code/phpdocs-basics#variadic-functions